arch/x86/core/userspace.S

```diff
@@ -169,6 +169,19 @@ SECTION_FUNC(TEXT, _x86_userspace_enter)
 	 */
 	mov	%esp, %edi
 
+#ifdef CONFIG_EXECUTION_BENCHMARKING
```

Comment:
You might be able to get away with moving this right before the "iret" since the code before it just formats the stack.

Reply:
Sure I can move it.